cluster-autoscaler/cloudprovider/clusterapi/clusterapi_unstructured.go

```diff
@@ -211,8 +211,8 @@ func (r unstructuredScalableResource) Taints() []apiv1.Taint {
 	}
 	if found {
 		for _, t := range newtaints {
-			if v, ok := t.(apiv1.Taint); ok {
-				taints = append(taints, v)
+			if t := unstructuredToTaint(t); t != nil {
+				taints = append(taints, *t)
 			} else {
 				klog.Warning("Unable to convert data to taint: %v", t)
```

Comment:
To make it easier to debug future casting issues, can we include `%T` in this to [format the type][1]?  Something like:

```go
klog.Warning("Unable to convert %T to taint: %v", t, t)
```

[1]: https://pkg.go.dev/fmt#hdr-Printing